nexus/src/app/mod.rs

```diff
@@ -1344,14 +1344,33 @@ async fn switch_zone_address_mappings(
 async fn map_switch_zone_addrs(
     log: &Logger,
     switch_zone_addresses: Vec<Ipv6Addr>,
+    resolver: &internal_dns_resolver::Resolver,
 ) -> HashMap<SwitchLocation, Ipv6Addr> {
     use gateway_client::Client as MgsClient;
     info!(log, "Determining switch slots managed by switch zones");
     let mut switch_zone_addrs = HashMap::new();
 
     for addr in switch_zone_addresses {
+        let port = match resolver
```

Comment:
Special scrutiny is warranted here. I think this should be ok as we fallback to the previous constant port when DNS does not work out.